x/ibc/core/exported/client.go

```diff
@@ -181,3 +197,12 @@ type Height interface {
 	Decrement() (Height, bool)
 	String() string
 }
+
+// GenesisMetadata is a wrapper interface over clienttypes.GenesisMetadata
+// all clients must use the concrete implementation in types
+type GenesisMetadata interface {
```

Comment:
So I think we should add a ClientType function to ensure we are not storing metadata from unsupported clients on genesis